### PR TITLE
nixos/nginx: add gitweb sub-service

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

Similarly to `services.lighttpd.gitweb`. Should be a better match for `nginx` than `cgit`, since it supports `FastCGI` natively.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/services/web-servers/nginx/gitweb.nix [hunk 1]
```diff
@@ -0,0 +1,91 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.nginx.gitweb;
+  gitwebConfigFile = pkgs.writeText "gitweb.conf" ''
```
Comment (wmertens) [inline nixos/modules/services/web-servers/nginx/gitweb.nix]: Could this not be shared with the lighttpd version? In a project of this size, DRY is very important, so updates apply everywhere…

**Review by wmertens — CHANGES_REQUESTED** [review]
Looks good, but I think it would be better for the gitweb config to stand on its own, perhaps in git-and-tools…

File: nixos/modules/services/web-servers/nginx/gitweb.nix (hunk 1 above)
Comment (gnidorah) [reply nixos/modules/services/web-servers/nginx/gitweb.nix]: @wmertens Done, thanks!

**Comment by wmertens** [comment]
👌

**Review by wmertens — APPROVED** [review]

**Comment by gnidorah** [comment]
@wmertens Thanks!

**Pull request merged**